### PR TITLE
Super market

### PR DESCRIPTION
**Pull request by sunnyvempati**
create market of markets.

### CODE REVIEW

File: markets/perps-market/contracts/interfaces/IPerpsMarketModule.sol [hunk 1]
```diff
@@ -7,6 +7,11 @@ import {AsyncOrder} from "../storage/AsyncOrder.sol";
  * @title Perps market module
  */
 interface IPerpsMarketModule {
+    event MarketOwnerNominated(uint128 indexed perpsMarketId, address newNominatedOwner);
+    event MarketOwnerChanged(uint128 indexed perpsMarketId, address oldOwner, address newOwner);
+
+    error NotNominated(address notNominatedAddress);
+
```
Comment (sunnyvempati) [inline markets/perps-market/contracts/interfaces/IPerpsMarketModule.sol:14]: can remove market owner

File: markets/perps-market/contracts/modules/AsyncOrderModule.sol [hunk 2]
```diff
@@ -200,31 +200,30 @@ contract AsyncOrderModule is IAsyncOrderModule {
             perpsAccount.addCollateralAmount(SNX_USD_MARKET_ID, runtime.pnlUint);
```
Comment (sunnyvempati) [inline markets/perps-market/contracts/modules/AsyncOrderModule.sol:200]: fix this

File: markets/perps-market/contracts/storage/PerpsAccount.sol [hunk 3]
```diff
@@ -55,6 +55,16 @@ library PerpsAccount {
         }
     }
 
+    /**
+        @notice allows us to update the account id in case it needs to be
+     */
+    function loadUpdate(uint128 id) internal returns (Data storage account) {
```
Comment (sunnyvempati) [inline markets/perps-market/contracts/storage/PerpsAccount.sol:61]: change to `create`

File: markets/perps-market/contracts/storage/PerpsAccount.sol [hunk 4]
```diff
@@ -218,18 +222,29 @@ library PerpsAccount {
...
-                (uint amountSold, ) = PerpsMarketFactory.load().spotMarket.sellExactIn(
+                (uint amountUsd, ) = PerpsMarketFactory.load().spotMarket.sellExactIn(
                     synthMarketId,
                     amount,
                     0,
                     address(0)
                 );
-                self.collateralAmounts[SNX_USD_MARKET_ID] += amountSold;
+                factory.synthetix.depositMarketUsd(factory.perpsMarketId, amountUsd);
+                self.collateralAmounts[synthMarketId] = 0;
```
Comment (sunnyvempati) [inline markets/perps-market/contracts/storage/PerpsAccount.sol:247]: need to update global?

Comment (sunnyvempati) [reply markets/perps-market/contracts/storage/PerpsAccount.sol:247]: call updateCollateralAmount with -amount

File: markets/perps-market/contracts/modules/PerpsMarketFactoryModule.sol [hunk 5]
```diff
@@ -56,40 +58,56 @@ contract PerpsMarketFactoryModule is IPerpsMarketFactoryModule {
...
+
     /**
      * @inheritdoc IPerpsMarketFactoryModule
      */
     function createMarket(
+        uint128 requestedMarketId,
         string memory marketName,
         string memory marketSymbol,
         address marketOwner
     ) external override returns (uint128) {
```
Comment (leomassazza) [inline markets/perps-market/contracts/modules/PerpsMarketFactoryModule.sol:85]: Anybody can create a market? (I don't find an access control)

File: markets/perps-market/contracts/storage/PerpsAccount.sol [hunk 6]
```diff
@@ -276,7 +290,7 @@ library PerpsAccount {
                         type(uint).max,
                         address(0)
```
Comment (leomassazza) [inline markets/perps-market/contracts/storage/PerpsAccount.sol:291]: same for referer here.

File: markets/perps-market/contracts/interfaces/IPerpsMarketFactoryModule.sol [hunk 7]
```diff
@@ -6,35 +6,37 @@ import {ISynthetixSystem} from "./external/ISynthetixSystem.sol";
...
 interface IPerpsMarketFactoryModule is IMarket {
-    event MarketRegistered(
+    event FactoryInitialized(uint128 globalPerpsMarketId);
+    event MarketCreated(
         uint128 indexed perpsMarketId,
         address indexed marketOwner,
         string marketName,
         string marketSymbol
     );
-    event MarketOwnerNominated(uint128 indexed perpsMarketId, address newNominatedOwner);
```
Comment (sunnyvempati) [inline markets/perps-market/contracts/interfaces/IPerpsMarketFactoryModule.sol:15]: remove

File: markets/perps-market/contracts/storage/GlobalPerpsMarket.sol [hunk 8]
```diff
@@ -39,6 +46,37 @@ library GlobalPerpsMarket {
...
     }
 
+    function totalCollateralValue(Data storage self) internal view returns (uint256 total) {
+        ISpotMarketSystem spotMarket = PerpsMarketFactory.load().spotMarket;
+        SetUtil.UintSet storage activeCollateralTypes = self.activeCollateralTypes;
+        uint256 activeCollateralLength = activeCollateralTypes.length();
+        for (uint i = 1; i < activeCollateralLength; i++) {
+            uint128 synthMarketId = activeCollateralTypes.valueAt(i).to128();
+
+            (uint collateralValue, ) = spotMarket.quoteSellExactIn(
```
Comment (bachstatter) [inline markets/perps-market/contracts/storage/GlobalPerpsMarket.sol:56]: If snxUSD dont quote sell, just grab the amount directly and add to total

File: markets/perps-market/contracts/modules/PerpsMarketFactoryModule.sol [hunk 9]
```diff
@@ -99,60 +116,38 @@ contract PerpsMarketFactoryModule is IPerpsMarketFactoryModule {
...
-        market.nominatedOwner = address(0);
-    }
-
-    /**
-     * @inheritdoc IPerpsMarketFactoryModule
-     */
-    function getMarketOwner(uint128 perpsMarketId) external view override returns (address) {
-        return PerpsMarket.load(perpsMarketId).owner;
-    }
+    // /**
```
Comment (bachstatter) [inline markets/perps-market/contracts/modules/PerpsMarketFactoryModule.sol:119]: Some more comments to remove

**Review by bachstatter — APPROVED** [review]

**Pull request merged**